website/content/docs/ecs/terraform/secure-configuration.mdx

```diff
@@ -16,13 +16,45 @@ A secure Consul cluster should include the following:
 
 -> **NOTE:** This page assumes that you have already configured your Consul server with the above features.
 
+## Auth Method
+
+Consul on ECS uses the [AWS IAM Auth Method](/docs/security/acl/auth-methods/aws-iam) to enable ECS
+tasks to automatically obtain Consul ACL tokens during startup. See the [Architecture](/docs/ecs/architecture)
+page for an overview of task startup, task shutdown, ACL tokens, and the AWS IAM auth method for Consul on ECS.
+
+### Auth Method Configuration
+
+When using the Terraform modules for Consul on ECS, the auth method is automatically configured
+by the ACL controller, and ECS tasks will be configured by the `mesh-task` module to be able
+to log in to the auth method.
+
+The `mesh-task` module configures the task IAM role as follows:
+
+* Adds a scoped `iam:GetRole` permission, enabling the role to fetch details about itself.
+* Tags the IAM role with the service name
+* <EnterpriseAlert inline /> Tags the IAM role with the Consul Enterprise namespace
+
+The `acl-controller` module deploys the ACL controller. At startup, the ACL controller
+configures two instances of the AWS IAM auth method:
+
+* One instance of the AWS IAM auth method is configured for client tokens:
+  * A Consul role is created with `node:write` permissions for client tokens used by Consul clients.
+  * A binding rule is configured that attaches the Consul role to each token created during a successful login to this auth method instance.
+* One instance of the AWS IAM auth method is configured for service tokens:
+  * A binding rule attaches a [service identity](/docs/security/acl#service-identities) to each token created during a successful login to this auth method instance. The service name for this service identity is taken from the tag, `consul.hashicorp.com.service-name`, on the IAM role.
+  * <EnterpriseAlert inline /> A namespace binding rule creates the service token in the namespace specified in the tag, <code>consul.hashicorp.com.namespace</code>, on the IAM role.
+* Both instances of the auth method only permit IAM roles with an ARN matching this pattern, `arn:aws:iam::<ACCOUNT>:role/consul-ecs/*`. This allows IAM roles at the path `/consul-ecs/` to log in. The role path can be changed by setting the `iam_role_path` input variable for the `mesh-task` and `acl-controller` modules.
+
```

Comment:
This description seems to make more sense in the architecture section (and I think a lot of the content is already there?). For this page I think we should focus on what the user needs to do. So the last part about the IAM role needing to have a specific arn is important.

Maybe we can split this between "bringing your own task role" and "using the default task role"?